prototype_source/ios_gpu_workflow.rst

```diff
@@ -0,0 +1,108 @@
+
+
+.. code:: shell
+
+    ['aten::adaptive_avg_pool2d', 
+    'aten::add.Tensor', 
+    'aten::addmm', 
+    'aten::reshape', 
+    'aten::size.int', 
+    'metal::copy_to_host', 
+    'metal_prepack::conv2d_run']
+
+Those are all the ops we need to run the mobilenetv2 model on iOS GPU. Cool! Now that you have the ``mobilenetv2_metal.pt`` saved on your disk, let's move on to the iOS part.
+
+
+Use C++ APIs
+---------------------
+
+In this section, we'll be using the `HelloWorld example <https://github.com/pytorch/ios-demo-app>`_ to demonstrate how to use the C++ APIs. The first thing we need to do is to build a custom LibTorch from Source. Make sure you have deleted the **build** folder from the previous step in PyTorch root directory. Then run the command below
+
+.. code:: shell
+    
+    IOS_ARCH=arm64 USE_PYTORCH_METAL=1 ./scripts/build_ios.sh
+
+Note ``IOS_ARCH`` tells the script to build a arm64 version of Libtorch. This is because in PyTorch, Metal is only available for the iOS devices that support the Apple A9 chip or above. Once the build finished, follow the `Build PyTorch iOS libraries from source <https://pytorch.org/mobile/ios/#build-pytorch-ios-libraries-from-source>`_ section from the iOS tutorial to setup the XCode settings properly. Don't forget to copy the `./mobilenetv2_metal.pt` to your XCode project.
+
+Next we need to make some changes in ``TorchModule.mm``
+
+.. code:: objective-c
+    
+    - (NSArray<NSNumber*>*)predictImage:(void*)imageBuffer {
+      torch::jit::GraphOptimizerEnabledGuard opguard(false);
+      at::Tensor tensor = torch::from_blob(imageBuffer, {1, 3, 224, 224}, at::kFloat).metal();
+      auto outputTensor = _impl.forward({tensor}).toTensor().cpu();
+      ...
+      return nil;
+    }
+
+As you can see, we simply just call ``.metal()`` to move our input tensor from CPU to GPU, and then call ``.cpu()`` to move the result back. Internally, ``.metal()`` will copy the input data from the CPU buffer to a GPU buffer with a GPU compatible memory format. When `.cpu()` is invoked, the GPU command buffer will be flushed and synced. After `forward` finished, the final result will then be copied back from the GPU buffer back to a CPU buffer.
+
+The last step we have to do is to add the `Accelerate.framework` and the `MetalShaderPerformance.framework` to your xcode project.
+
+If everything works fine, you should be able to see the inference results on your phone. The result below was captured from an iPhone11 device
+
+.. code:: shell
+
+    - timber wolf, grey wolf, gray wolf, Canis lupus
+    - malamute, malemute, Alaskan malamute
+    - Eskimo dog, husky
+
+You may notice that the results are slighly different from the `results <https://pytorch.org/mobile/ios/#install-libtorch-via-cocoapods>`_ we got from the CPU model as shown in the iOS tutorial. This is because by default Metal uses fp16 rather than fp32 to compute. The precision loss is expected.
+
```

Comment:
Should we point to the GPU performance gain results at https://github.com/pytorch/pytorch/pull/46600 or do a quick summary here?